src/main/java/io/github/thebusybiscuit/slimefun4/utils/blockpattern/TShapedBlockPattern.java

```diff
@@ -0,0 +1,131 @@
+package io.github.thebusybiscuit.slimefun4.utils.blockpattern;
+
+import org.apache.commons.lang.Validate;
+import org.bukkit.Location;
+import org.bukkit.Material;
+import org.bukkit.block.Block;
+import org.bukkit.block.BlockFace;
+
+import javax.annotation.Nonnull;
+import javax.annotation.ParametersAreNonnullByDefault;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+
+/**
+ * This utility class holds methods relating to a pattern of blocks in a T-shape
+ *
+ * @author md5sha256
+ */
+@ParametersAreNonnullByDefault
+public class TShapedBlockPattern {
+
+    private TShapedBlockPattern() {}
+
+    /**
+     * Get the blocks of a specified material which are in a T-shape pattern at a given location.
+     * <br>
+     * <strong>NOTE: The T-Shape is defined as 3 blocks wide and 2 blocks tall</strong>
+     * <br>
+     * This method will always check in the east-west direction before the north-south direction.
+     * The reasoning behind east-west first is to match vanilla when iron golems or withers are spawned and
+     * both directions contain a valid T-shape, the east-west direction is always preferred.
+     *
+     * @param material The {@link Material} which the blocks should match
+     * @param location The {@link Location} of the lowest block at the center, aka the base block of the T-shape
+     * @return Returns non-null {@link Collection} of {@link Block}s which match the specified {@link Material} and are
+     * in a T-Shape in no particular order
+     */
+    public static @Nonnull Collection<Block> getMatchingBlocks(Material material, Location location) {
+        Validate.notNull(material, "Material cannot be null");
+        Validate.notNull(location, "Location cannot be null");
+        Validate.notNull(location.getWorld(), "Location#getWorld cannot be null");
```

Comment:
Use Preconditions as Validate only affects tests.

Reply:
Does validate only affect tests? Looking at the src it just throws regardless? I'll change it anyway but im a bit confused